privacyidea/models.py

```diff
@@ -372,6 +372,9 @@ def set_realms(self, realms, add=False):
         # add new TokenRealms
         # We must not set the same realm more than once...
         # uniquify: realms -> set(realms)
+        if self.first_owner:
+            if self.first_owner.realm.name not in realms:
+                realms.append(self.first_owner.realm.name)
```

Comment:
We had a little talk about this and we should issue an info-log message here that the realm of the assigned user was to be deleted.